online-shop/src/app/services/shopping-cart.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { ProductDetail } from '../modules/shared/types/products.types';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ShoppingCartService {
+  addProduct(product: ProductDetail) {
+    const itExists = JSON.parse(localStorage.getItem('itExists') || '{}') || [];
+    console.log(itExists);
+    itExists.push();
+    localStorage.setItem('2', JSON.stringify(itExists));
```

Comment:
Not quite sure what is the logic behind this, but I would assume that it's not finished yet? It doesn't seem to have any usage to me

Reply:
it was not finished yet; I have changed it into something functional now.